src/frontend/src/components/pinInput.test.ts

```diff
@@ -16,13 +16,13 @@ function dispatchInput(elem: HTMLInputElement, text: string) {
 
 // Dispatch an "ClipboardEvent"-ish event that works in jsdom
 function dispatchPaste(elem: HTMLInputElement, text: string) {
-  const event: any = new Event("paste", { bubbles: true, cancelable: true });
-  event.clipboardData = {
+  const evnt: any = new Event("paste", { bubbles: true, cancelable: true });
```

Comment:
not related: `any`?

Reply:
had to force it to accept `.clipboardData`, which TypeScript doesn't know about

Reply:
@peterpeterparker the following is a tad safer, do you prefer that?

```
// Dispatch an "ClipboardEvent"-ish event that works in jsdom
function dispatchPaste(elem: HTMLInputElement, text: string) {
  const event = new Event("paste", { bubbles: true, cancelable: true });
  (event as any).clipboardData = {
    getData() {
      return text;
    },
  };
  elem.dispatchEvent(event);
}
```

Reply:
`as ClipboardEvent` ?

Reply:
or `as unknown as ClipboardEvent`?

Reply:
ok